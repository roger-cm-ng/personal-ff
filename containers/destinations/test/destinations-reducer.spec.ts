import destinationsReducer from '../destinations-reducer';
import {
  DESTINATIONS_CONSTRUCTED,
  PLANET_CHANGED,
  VEHICLE_CHANGED
} from './../destinations-actions';
import {
  DESTINATIONS_INIT,
  DESTINATIONS_EMPTY_TO_DONLON,
  DESTINATIONS_DONLON_TO_ENCHAI,
  DESTINATIONS_EMPTY_TO_SPACE_POD,
  DESTINATIONS_SPACE_POD_TO_SPACE_ROCKET
} from './destinations-mock-data';

describe('destinations reducer', () => {
  const initialState = () => destinationsReducer([], {
    type: DESTINATIONS_CONSTRUCTED,
    payload: []
  });

  const constructedState = () => destinationsReducer(initialState(), {
    type: DESTINATIONS_CONSTRUCTED,
    payload: DESTINATIONS_INIT
  });

  const emptyToDonlonState = () => destinationsReducer(constructedState(), {
    type: PLANET_CHANGED,
    payload: {
      destination: 0,
      planet: {
        name: 'Donlon',
        distance: 100,
        isChosen: true
      }
    }
  });

  const donlonToEnchaiState = () => destinationsReducer(emptyToDonlonState(), {
    type: PLANET_CHANGED,
    payload: {
      destination: 0,
      planet: {
        name: 'Enchai',
        distance: 200,
        isChosen: true
      }
    }
  });

  const emptyToSpacePodState = () => destinationsReducer(donlonToEnchaiState(), {
    type: VEHICLE_CHANGED,
    payload: {
      destination: 0,
      prevVehicle: '',
      currentVehicle: {
        name: 'Space pod',
        total_no: 2,
        max_distance: 200,
        speed: 2,
        capacity: true,
        usage: 1
      }
    }
  });

  const spacePodToSpaceRocketState = () => destinationsReducer(emptyToSpacePodState(), {
    type: 'VEHICLE_CHANGED',
    payload: {
      destination: 0,
      prevVehicle: 'Space pod',
      currentVehicle: {
        name: 'Space rocket',
        total_no: 1,
        max_distance: 300,
        speed: 4,
        capacity: true,
        usage: 1
      }
    }
  });

  it('should return initial state', () => {
    expect(initialState()).toHaveLength(0);
  });

  it('should update the state with constructed data when type is DESTINATIONS_CONSTRUCTED', () => {
    expect(constructedState()).toMatchSnapshot();
    expect(constructedState()).toEqual(DESTINATIONS_INIT);
  });

  it('should update the state with planet Donlon when type is PLANET_CHANGED', () => {
    expect(emptyToDonlonState()).toEqual(DESTINATIONS_EMPTY_TO_DONLON);
  });

  it('should update the state with planet Enchai when type is PLANET_CHANGED', () => {
    expect(donlonToEnchaiState()).toEqual(DESTINATIONS_DONLON_TO_ENCHAI);
  });

  it('should update the state with vehicle Space pod when type is VEHICLE_CHANGED', () => {
    expect(emptyToSpacePodState()).toEqual(DESTINATIONS_EMPTY_TO_SPACE_POD);
  });

  it('should update the state with vehicle Space rocket when type is VEHICLE_CHANGED', () => {
    expect(spacePodToSpaceRocketState()).toEqual(DESTINATIONS_SPACE_POD_TO_SPACE_ROCKET);
  });
});
