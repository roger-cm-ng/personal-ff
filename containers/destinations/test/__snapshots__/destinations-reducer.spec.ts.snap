// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`destinations reducer should update the state with constructed data when type is DESTINATIONS_CONSTRUCTED 1`] = `
Array [
  Object {
    "planet": Object {
      "distance": 0,
      "isChosen": false,
      "name": "",
    },
    "vehicle": Object {
      "capacity": false,
      "max_distance": 0,
      "name": "",
      "speed": 0,
      "total_no": 0,
      "usage": 0,
    },
    "vehiclesInstance": Array [
      Object {
        "capacity": true,
        "max_distance": 200,
        "name": "Space pod",
        "speed": 2,
        "total_no": 2,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 300,
        "name": "Space rocket",
        "speed": 4,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 400,
        "name": "Space shuttle",
        "speed": 5,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 600,
        "name": "Space ship",
        "speed": 10,
        "total_no": 2,
        "usage": 0,
      },
    ],
  },
  Object {
    "planet": Object {
      "distance": 0,
      "isChosen": false,
      "name": "",
    },
    "vehicle": Object {
      "capacity": false,
      "max_distance": 0,
      "name": "",
      "speed": 0,
      "total_no": 0,
      "usage": 0,
    },
    "vehiclesInstance": Array [
      Object {
        "capacity": true,
        "max_distance": 200,
        "name": "Space pod",
        "speed": 2,
        "total_no": 2,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 300,
        "name": "Space rocket",
        "speed": 4,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 400,
        "name": "Space shuttle",
        "speed": 5,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 600,
        "name": "Space ship",
        "speed": 10,
        "total_no": 2,
        "usage": 0,
      },
    ],
  },
  Object {
    "planet": Object {
      "distance": 0,
      "isChosen": false,
      "name": "",
    },
    "vehicle": Object {
      "capacity": false,
      "max_distance": 0,
      "name": "",
      "speed": 0,
      "total_no": 0,
      "usage": 0,
    },
    "vehiclesInstance": Array [
      Object {
        "capacity": true,
        "max_distance": 200,
        "name": "Space pod",
        "speed": 2,
        "total_no": 2,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 300,
        "name": "Space rocket",
        "speed": 4,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 400,
        "name": "Space shuttle",
        "speed": 5,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 600,
        "name": "Space ship",
        "speed": 10,
        "total_no": 2,
        "usage": 0,
      },
    ],
  },
  Object {
    "planet": Object {
      "distance": 0,
      "isChosen": false,
      "name": "",
    },
    "vehicle": Object {
      "capacity": false,
      "max_distance": 0,
      "name": "",
      "speed": 0,
      "total_no": 0,
      "usage": 0,
    },
    "vehiclesInstance": Array [
      Object {
        "capacity": true,
        "max_distance": 200,
        "name": "Space pod",
        "speed": 2,
        "total_no": 2,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 300,
        "name": "Space rocket",
        "speed": 4,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 400,
        "name": "Space shuttle",
        "speed": 5,
        "total_no": 1,
        "usage": 0,
      },
      Object {
        "capacity": true,
        "max_distance": 600,
        "name": "Space ship",
        "speed": 10,
        "total_no": 2,
        "usage": 0,
      },
    ],
  },
]
`;
